### PR TITLE
Voice notes

### PR DESCRIPTION
**Pull request by moxie0**
// FREEBIE

### CODE REVIEW

File: src/org/thoughtcrime/securesms/ConversationActivity.java [hunk 1]
```diff
@@ -1344,6 +1370,67 @@ public void onCameraFail() {
...
+  @Override
+  public void onRecorderFinished() {
+    Vibrator vibrator = (Vibrator)getSystemService(Context.VIBRATOR_SERVICE);
+    vibrator.vibrate(20);
+
+    ListenableFuture<Pair<Uri, Long>> future = audioRecorder.stopRecording();
+    future.addListener(new AssertedSuccessListener<Pair<Uri, Long>>() {
+      @Override
+      public void onSuccess(@Nullable final Pair<Uri, Long> result) {
+        if (result == null) {
```
Comment (mcginty) [inline src/org/thoughtcrime/securesms/ConversationActivity.java]: reads a little weird that this is an AssertedSuccessListener but we check for a failure case onSuccess

File: src/org/thoughtcrime/securesms/ConversationActivity.java [hunk 2]
```diff
@@ -1344,6 +1370,67 @@ public void onCameraFail() {
...
+        try {
+          boolean    forceSms   = sendButton.isManualSelection() && sendButton.getSelectedTransport().isSms();
+          AudioSlide audioSlide = new AudioSlide(ConversationActivity.this, result.first, result.second);
+          SlideDeck  slideDeck  = new SlideDeck();
+          slideDeck.addSlide(audioSlide);
+
+          sendMediaMessage(forceSms, "", slideDeck).addListener(new AssertedSuccessListener<Void>() {
+            @Override
+            public void onSuccess(Void nothing) {
+              PersistentBlobProvider.getInstance(ConversationActivity.this).delete(result.first);
```
Comment (mcginty) [inline src/org/thoughtcrime/securesms/ConversationActivity.java]: is this run off the main thread?

File: src/org/thoughtcrime/securesms/ConversationActivity.java [hunk 3]
```diff
@@ -1344,6 +1370,67 @@ public void onCameraFail() {
...
+  public void onRecorderCanceled() {
+    Vibrator vibrator = (Vibrator)getSystemService(Context.VIBRATOR_SERVICE);
+    vibrator.vibrate(50);
+
+    ListenableFuture<Pair<Uri, Long>> future = audioRecorder.stopRecording();
+
+    future.addListener(new AssertedSuccessListener<Pair<Uri, Long>>() {
+      @Override
+      public void onSuccess(@Nullable Pair<Uri, Long> result) {
+        if (result != null) PersistentBlobProvider.getInstance(ConversationActivity.this).delete(result.first);
```
Comment (mcginty) [inline src/org/thoughtcrime/securesms/ConversationActivity.java]: same question about asserting this is run off the main thread

File: src/org/thoughtcrime/securesms/util/AnimationUtil.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
+package org.thoughtcrime.securesms.util;
+
+import android.view.View;
+import android.view.animation.AlphaAnimation;
+import android.view.animation.Animation;
+
+public class AnimationUtil {
+
+  public static void fadeOut(View view, int duration) {
```
Comment (mcginty) [inline src/org/thoughtcrime/securesms/util/AnimationUtil.java]: these methods already exist in ViewUtil

**Comment by mcginty** [comment]
seems like a large chunk of these animation sets can be done in xml

**Comment by moxie0** [comment]
updated

**Pull request closed without merging**